### ISSUE TITLE
Tests failing on macOS

### ISSUE THREAD

**Issue by jlingema** [HEP-FCC/podio#56]
While testing our packages on macOS for HEP-FCC/fcc-edm#49 I noticed that the tests for podio fail because the podio library can't be found. 

It looks like now we need to set `DYLD_LIBRARY_PATH`, e.g. [here](https://github.com/HEP-FCC/podio/blob/master/tests/CMakeLists.txt#L48). Any ideas why this could have changed recently?

**Comment by cbernet** [comment]
No idea… 

Not sure about the test suite, but for interactive use, the DYLD lib path is set in the init.sh
https://github.com/HEP-FCC/podio/blob/master/init.sh <https://github.com/HEP-FCC/podio/blob/master/init.sh>
Maybe this script is not sourced correctly?

**Comment by jlingema** [comment]
The bizarre thing is that it works interactively without a problem. Also inspecting DYLD library path shows that everything is fine. It's just the test suite that fails.

**Comment by cbernet** [comment]
I don’t know if it could be related but in macos sierra, the SIP wipes out this env var in child shells… 
I had to turn off SIP to be able to use our software.

**Comment by jlingema** [comment]
Indeed, turning off SIP fixes the problem. For future reference, I followed [this guide](https://developer.apple.com/library/content/documentation/Security/Conceptual/System_Integrity_Protection_Guide/ConfiguringSystemIntegrityProtection/ConfiguringSystemIntegrityProtection.html).

But turning the whole protection off seems a bit drastic? I'll try to find out if there's a less invasive way of dealing with this.

**Comment by cbernet** [comment]
Dunno, I think it is the protection that is a bit drastic actually! :-)
But if you can find a solution, let me know.

**Comment by jlingema** [comment]
From a quick google search it would seem that the standard way of dealing with this is to enable the RPATH for macOS builds by adding `-DCMAKE_MACOSX_RPATH=ON` (as the CMake warnings actually point out :-|), which indeed works on my system after re-enabling the system integrity protection. We could add this to the CMakeLists.

**Comment by zaborowska** [comment]
Hi @cbernet, can you comment on that issue?
